rad/src/lib.rs

```diff
@@ -138,6 +141,19 @@ fn rng_response_with_data_report(
     Ok(RadonReport::from_result(Ok(result), context))
 }
 
+/// Handle HTTP-POST response with data report
+fn http_post_response_with_data_report(
+    retrieve: &RADRetrieve,
+    response: &str,
+    context: &mut ReportContext<RadonTypes>,
+    settings: RadonScriptExecutionSettings,
+) -> Result<RadonReport<RadonTypes>> {
+    let input = RadonTypes::from(RadonString::from(response));
+    let radon_script = unpack_radon_script(&retrieve.script)?;
+
+    execute_radon_script(input, &radon_script, context, settings)
+}
+
```

Comment:
should we rename to http_response_with_data_report and do not have the same function with 2 names?